### PR TITLE
PR - ⭐ : Add aks feature

### PR DESCRIPTION
**Pull request by venkatsunilm**
1. Add aks feature
2. Modified readme file
3. Add few changes to common and core classes as well
4. Renamed RG, storage and key valult names to maintain consitency
5. Add documentation
6. Follow few best practices

### CODE REVIEW

File: tf/aks/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  "isDefault": true,
+  "managedByTenants": [],
+  "name": "Azure subscription venkat test",
+  "state": "Enabled",
+  "tenantId": "3cc81ecd06f86",
+  "user": {
+    "name": "ve...@hotmail.com",
+    "type": "user"
+  }
+}
```
Comment (AymenSegni) [inline tf/aks/README.md:28]: Please don't add your tenant details (IDs) into git files, you can change it with dummy values XXXXXXXXXX

File: tf/aks/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  "clientSecret": "wRb8QF-8pyav7",
+  "subscriptionId": "0cea2519ddb",
+  "tenantId": "3cc81ecd06f86",
+  "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
+  "resourceManagerEndpointUrl": "https://management.azure.com/",
+  "activeDirectoryGraphResourceId": "https://graph.windows.net/",
+  "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
+  "galleryEndpointUrl": "https://gallery.azure.com/",
+  "managementEndpointUrl": "https://management.core.windows.net/"
+}
```
Comment (AymenSegni) [inline tf/aks/README.md:40]: It is crucial that you do not share your Azure SP credentials, including IDs and secrets, in any git files. This information should be kept private to ensure the security of your account and data. Please replace them with dummy values

File: tf/aks/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+output "resource_group_name" {
+  value       = module.aks.resource_group_name
+  description = "The name of the resource group where the AKS cluster is deployed."
+}
+
+output "aks_client_key" {
+  value       = module.aks.aks_client_key
+  description = "The client key used for authenticating with the AKS cluster."
+}
```
Comment (AymenSegni) [inline tf/aks/outputs.tf:31]: This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

File: tf/aks/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+output "aks_client_certificate" {
+  value       = module.aks.aks_client_certificate
+  description = "The client certificate used for authenticating with the AKS cluster."
+}
```
Comment (AymenSegni) [inline tf/aks/outputs.tf:37]: This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

File: tf/aks/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+output "aks_cluster_ca_certificate" {
+  value       = module.aks.aks_cluster_ca_certificate
+  description = "The CA (Certificate Authority) certificate used by the AKS cluster."
+}
```
Comment (AymenSegni) [inline tf/aks/outputs.tf:43]: This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

File: tf/aks/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+output "aks_cluster_username" {
```
Comment (AymenSegni) [inline tf/aks/outputs.tf:45]: This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

File: tf/aks/outputs.tf [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  value       = module.aks.aks_cluster_username
+  description = "The username for authenticating with the AKS cluster."
+}
+
+output "aks_cluster_password" {
+  value       = module.aks.aks_cluster_password
+  description = "The password for authenticating with the AKS cluster."
+}
```
Comment (AymenSegni) [inline tf/aks/outputs.tf:55]: This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

File: tf/aks/variables.tf [hunk 8]
```diff
@@ -0,0 +1,29 @@
+variable "environment" {
+  description = "Name of the environment"
+  type        = string
+}
```
Comment (AymenSegni) [inline tf/aks/variables.tf:11]: default is missing

File: tf/aks/variables.tf [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+variable "location" {
+  description = "Azure location to use"
+  type        = string
+}
```
Comment (AymenSegni) [inline tf/aks/variables.tf:17]: default is missing

File: tf/aks/variables.tf [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+variable "kubernetes_version" {
+  description = "The Kubernetes version to use"
+  type        = string
+}
```
Comment (AymenSegni) [inline tf/aks/variables.tf:23]: default is missing

File: tf/aks/variables.tf [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+variable "service_principal_client_id" {
+  description = "The client ID of the service principal to be used by AKS"
+  type        = string
+}
```
Comment (AymenSegni) [inline tf/aks/variables.tf:30]: default is missing

File: tf/aks/variables.tf [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+variable "service_principal_client_secret" {
+  description = "The client secret of the service principal to be used by AKS"
+  type        = string
+}
```
Comment (AymenSegni) [inline tf/aks/variables.tf:24]: default is missing

File: tf/aks/variables.tf [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+variable "service_principal_client_id" {
+  description = "The client ID of the service principal to be used by AKS"
+  type        = string
+}
+
+variable "service_principal_client_secret" {
+  description = "The client secret of the service principal to be used by AKS"
+  type        = string
```
Comment (AymenSegni) [inline tf/aks/variables.tf:34]: add sensitive = true

File: tf/aks/variables.tf [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+variable "ssh_public_key" {
+  description = "The SSH public key to use with AKS"
+  type        = string
+}
```
Comment (AymenSegni) [inline tf/aks/variables.tf:44]: add sensitive = true

File: tf/aks/README.md (hunk 1 above)
Comment (venkatsunilm) [reply tf/aks/README.md:28]: These are actually dummy values 😊, but yes i will replace them like you suggested.

File: tf/aks/README.md (hunk 2 above)
Comment (venkatsunilm) [reply tf/aks/README.md:40]: Sure again it's dummy values but then yes I understood it's better to replace with XXXXXXX. I will replace it. :)

File: tf/aks/outputs.tf [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+output "aks_kube_config" {
+  value       = module.aks.aks_kube_config
+  sensitive   = true
+  description = "The Kubernetes configuration file (kubeconfig) for accessing the AKS cluster."
+}
+
+output "aks_host" {
+  value       = module.aks.aks_host
+  description = "The host URL of the AKS cluster."
+}
```
Comment (venkatsunilm) [inline tf/aks/outputs.tf]: @AymenSegni Is this config and host are sensitive information?

File: tf/aks/README.md (hunk 1 above)
Comment (venkatsunilm) [reply tf/aks/README.md:28]: Replaced with dummy names.

File: tf/aks/README.md (hunk 2 above)
Comment (venkatsunilm) [reply tf/aks/README.md:40]: Replaced with dummy names and removed the unwanted info.

File: tf/aks/outputs.tf (hunk 15 above)
Comment (AymenSegni) [reply tf/aks/outputs.tf]: From the description: " description = "The password for authenticating with the AKS cluster."
It's very sensitive : )

Comment (venkatsunilm) [reply tf/aks/outputs.tf]: I was asking about the config and host not the password. :)

Comment (AymenSegni) [reply tf/aks/outputs.tf]: The host URL isn't sensitive, however config is

**Pull request merged**